### PR TITLE
ADO#4825 -- Create MSCA member selection screen

### PR DESCRIPTION
**Pull request by AdamP-HRSDC-RHDCC**
### Description
Created the member selection screen based on the behind MSCA Figma design

### Related Azure Boards Work Items
ADO#4825

### Screenshots (if applicable)
ms

### Additional Notes
TODO Link review pages to edit mode when available

### CODE REVIEW

File: frontend/app/.server/routes/helpers/protected-renew-route-helpers.ts [hunk 1]
```diff
@@ -13,6 +13,15 @@ import { getCdcpWebsiteApplyUrl } from '~/utils/url-utils.server';
 export interface ProtectedRenewState {
   readonly id: string;
   readonly editMode: boolean;
+  readonly memberSelection: boolean[];
+  readonly externallyReviewed: boolean[];
+  readonly previouslyReviewed: boolean[];
```
[the remark this answers is not in the document]
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/.server/routes/helpers/protected-renew-route-helpers.ts]: Nested these values inside the applicantInformation and children objects

File: frontend/app/routes/protected/renew/$id/member-selection.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            </ul>
+          </>
+        )}
+        <p className="mb-4">{t('protected-renew:member-selection.form-instructions')}</p>
+        <fetcher.Form method="post" noValidate>
+          <input type="hidden" name="_csrf" value={csrfToken} />
+          <div className="mt-6 space-y-8">
+            <section key={0} className="rounded border border-gray-500">
+              <label className="flex">
+                <input type="checkbox" id={`input-0`} name="member-0" onChange={handleOnSelect} className="hidden" />
```
[the remark this answers is not in the document]
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/routes/protected/renew/$id/member-selection.tsx]: Created a custom CardLink component for memberSelection based off the home.tsx component

File: frontend/app/routes/protected/renew/$id/member-selection.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+import { getTitleMetaTags } from '~/utils/seo-utils';
+
+enum FormAction {
+  Add = 'add',
+  Continue = 'continue',
+  Cancel = 'cancel',
+  Save = 'save',
+  Remove = 'remove',
+  Back = 'back',
+}
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/member-selection.tsx:34]: We can remove `Add` and `Remove` as they aren't being used.

File: frontend/app/routes/protected/renew/$id/member-selection.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                if (child.externallyReviewed) {
+                  const childName = `${child.firstName} ${child.lastName}`;
+                  return (
+                    <li key={childName} className="mb-4">
+                      {childName}
+                    </li>
+                  );
+                }
+                return <></>;
+              })}
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/member-selection.tsx:121]: Nitpick: We can use `filter(..)` to exclude the non-reviewed children and avoid the empty fragments `<></>`. We can also remove `index` parameter from the `map(..)` as it is not being used.

```suggestion
              {children
                .filter((child) => child.externallyReviewed)
                .map((child) => {
                  const childName = `${child.firstName} ${child.lastName}`;
                  return (
                    <li key={childName} className="mb-4">
                      {childName}
                    </li>
                  );
                })}
```

File: frontend/app/routes/protected/renew/$id/member-selection.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            </ul>
+          </>
+        )}
+        <p className="mb-4">{t('protected-renew:member-selection.form-instructions')}</p>
+        <fetcher.Form method="post" noValidate>
+          <input type="hidden" name="_csrf" value={csrfToken} />
+          <div className="mt-6 space-y-8">
+            <CardLink key={applicantName} title={applicantName} previouslyReviewed={applicant?.previouslyReviewed} routeId="protected/renew/$id/dental-insurance" params={params} />
+            {children.map((child, index) => {
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/member-selection.tsx:130]:
```suggestion
            {children.map((child) => {
```

**Review by nicholas-ly — COMMENTED** [review]
Made some minor comments. We'll eventually have to introduce another PR that calls `findClientApplicationBySin(..)` to display the `applicant` and `children` information and clean up the state object to only include fields modifiable by the user.

File: frontend/app/routes/protected/renew/$id/member-selection.tsx (hunk 3 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/routes/protected/renew/$id/member-selection.tsx]: Removed the unused enums

File: frontend/app/routes/protected/renew/$id/member-selection.tsx (hunk 5 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/routes/protected/renew/$id/member-selection.tsx]: Removed index vars

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**